src/filesystem/SDL_filesystem.c

```diff
@@ -100,3 +100,105 @@ int SDL_GetPathInfo(const char *path, SDL_PathInfo *info)
     }
     return SDL_SYS_GetPathInfo(path, info);
 }
+
+
+static SDL_bool WildcardMatch(const char *pattern, const char *str)
```

Comment:
Should we use the actual glibc glob function when it’s available?

Reply:
I worry that doing so means quirks are going to work on some targets and not others. For example, glob() ignores files that start with a '.' unless the pattern starts with a period too, because of Unix culture.

Also, the code to manage optionally building with glob() support would be larger than the WildcardMatch function.  :)